Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+  private static _repositionTimer: Nullable<number> = null;
+  private static _statusPopupDelay: number = 200;
+  private static _statusPopupRepositionTimer: Nullable<number> = null;
+  private static _blurTimer: Nullable<number> = null;
+  private static _updateFocus: boolean = true;
+
+  public static AddMenuInfo(menuInfo: DropDownMenu_MenuInfo): void
+  {
+    DropDownMenu._menuInfos[menuInfo.ID] = menuInfo;
+  }
+
+  public static BindOpenEvent (openTarget: JQuery, menuID: string, eventType: string, getSelectionCount: Nullable<DropDownMenu_SelectionCountGetter>, moveToMousePosition: boolean): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery ('openTarget', openTarget);
+    ArgumentUtility.CheckNotNullAndTypeIsString ('menuID', menuID);
+    ArgumentUtility.CheckNotNullAndTypeIsString ('eventType', eventType);
+    ArgumentUtility.CheckNotNullAndTypeIsBoolean ('moveToMousePosition', moveToMousePosition);
+
+    var menuButton = openTarget[0].tagName.toLowerCase() === 'a' ? openTarget : openTarget.find ('a[href]:last');
+    menuButton.attr('aria-haspopup', 'menu');
+    menuButton.attr('aria-expanded', 'false');
+
+    openTarget.bind(
+        eventType,
+        function (evt: JQueryMouseEventObject)
+        {
+          menuButton.focus();
+          DropDownMenu.OnClick (openTarget, menuID, getSelectionCount, moveToMousePosition ? evt : null);
+        });
+
+    if (!moveToMousePosition)
+    {
+      menuButton
+          .bind ("focus", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).addClass (DropDownMenu._focusClassName); })
+          .bind("blur", function (evt) { openTarget.find('.' + DropDownMenu._buttonClassName).removeClass (DropDownMenu._focusClassName); });
+
+      var allButMenuButton = openTarget.find ('a[href]').not(menuButton);
+      allButMenuButton
+          .bind ("mouseover", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).addClass (DropDownMenu._nestedHoverClassName); })
+          .bind ("mouseout", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).removeClass (DropDownMenu._nestedHoverClassName); });
+    }
+  }
+
+  public static LoadFilteredMenuItems(
+      itemInfos: DropDownMenu_ItemInfo[],
+      loadMenuItemStatus: Nullable<DropDownMenu_MenuInfo_LoadMenuItemsCallback>,
+      onSuccess: DropDownMenu_MenuInfo_LoadMenuItemsSuccessHandler,
+      onError: DropDownMenu_MenuInfo_LoadMenuItemsErrorHandler): void
+  {
+    if (loadMenuItemStatus !== null)
+    {
+      const currentLoadOperationID = new Object();
+      const onSuccessForLoadMenuItemStatus = function (itemStatusArray: DropDownMenu_ItemInfo[])
+      {
+        if (DropDownMenu._currentLoadOperation !== currentLoadOperationID)
+          return;
+        DropDownMenu._currentLoadOperation = null;
+
+        const itemStatusMap = itemStatusArray.reduce(
+          function (map: Dictionary<DropDownMenu_ItemInfo>, itemStatus: DropDownMenu_ItemInfo)
```

Comment:
ItemStatus should be a separate type, not DropDownMenu_ItemInfo. See WebMenuItemProxy